TeamAmaze/AmazeFileManager
# Access Android/data or obb dirs individually as app package

## Description
Instead of previously request for accessing the whole Android/data and Android/obb directories, change to ask for access to individual package directory under Android/{data,obb}.

Also changed display of directories under Android/{data,obb} to comparing with installed packages, and display only those matches installed apps.

Create directory or file at Android/data is not allowed with this changeset, with regards to updated access model in newer Androids, for an easy solution.

No upgrade concerns IMO, as granted URI permissions should have nothing to do with this changeset, since it is not using the URI to list directories under Android/data.

#### Issue tracker   
Fixes #3032 (perhaps?)

#### Automatic tests
- [x] Added test cases (partial, unit test only)
  
#### Manual tests

Devices:
- Nexus 5x running LineageOS 18 (Android 10)
- Fairphone 3 running LineageOS 19.1 (Android 12 SDK 32)  
- Pixel 6 emulator running Android 13 SDK33

The following test cases should succeed without problem.
1. Android/data access
- navigate into/out of Android/data should have no problem, without asking for explicit access
- should be able to list directories created by installed apps
2. Directories under Android/data
- when accessing directories under Android/data, prompt will popup brings up DocumentsUI with location set at app package directory under Android/data. User can grant permission to access that directory by pressing "Use this folder"
- File/Directory CRUD operations including cut/copy and paste should have no problem after granting access
- Navigate from directories under Android/data upward, to above Android/data, should succeed without problem
- Navigate from directories under Android/data upward, to other directory under Android/data, should succeed without problem
3. Regression: OTG access
- File/Directory CRUD operations when device is plugged with storage device over OTG should continue to work without problem
4. Regression: Amaze FTP server (built-in)
- If using SAF (e.g. using SD card as shared directory) at FTP server, file/directory CRUD operations should continue to work as FTP client interact with Amaze's FTP server

This changeset is targeted at API >= 30, but may also test on older devices for regressions if available.

#### Build tasks success  
Successfully running following tasks on local:
